src/utils/test_main.hxx

```diff
@@ -177,6 +178,49 @@ void run_x(std::function<void()> fn)
     g_executor.sync_run(std::move(fn));
 }
 
+/// Structure holding returned objects for an invoke_flow_nowait command.
+template <class T> struct PendingInvocation
+{
+    /// Buffer sent to the flow.
+    BufferPtr<T> b;
+    /// Notifiable to wait for.
+    SyncNotifiable notifiable;
+    /// Barrier notifiable given to the buffer.
+    BarrierNotifiable barrier {&notifiable};
+    /// True if wait has been invoked.
+    bool isWaited {false};
+
+    ~PendingInvocation()
+    {
+        wait();
+    }
+
+    void wait()
+    {
+        if (isWaited)
+        {
+            return;
+        }
+        notifiable.wait_for_notification();
+        isWaited = true;
+    }
+};
+
+/// Executes a callable flow similar to invoke_flow(...) but does not wait for
+/// the result to come back. Instead, returns a PendingInvocation object, where
+/// there is a wait() method to be called.
+template <class T, typename... Args>
+std::unique_ptr<PendingInvocation<T>> invoke_flow_nowait(
+    FlowInterface<Buffer<T>> *flow, Args &&...args)
+{
+    auto ret = std::make_unique<PendingInvocation<T>>();
```

Comment:
std::make_unique is only available in C++14 or later. Arduino-esp32 is pinned to C++11 (due to IDF using this).

Reply:
ack. you might want to reach out to the esp-idf folks on what their plans are. 11 is a long time ago now...

Reply:
A PR was submitted but it is not planned for merge until at least IDF v5.0: https://github.com/espressif/esp-idf/pull/4846

Reply:
I use C++14 in my projects without issues though..